### PR TITLE
Fix collection KMP by duplicating Lock

### PR DESCRIPTION
**Pull request by yigit**
This is just a temporary fix

Bug: n/a
Test: gw bOS

### CODE REVIEW

**Comment by dlam** [comment]
How did core fail here?

**Review by dlam — APPROVED** [review]

**Comment by dlam** [comment]
oh it only failed on push event, it seems unrelated anyhow, but I am confused on why there are no logs in the build step.

**Comment by dlam** [comment]
Also, should we file a bug to de-duplicate?

**Comment by yigit** [comment]
> Also, should we file a bug to de-duplicate?

Yesi should do that. Was quickly sending this to try before leaving home to commute.

File: collection/collection/src/macosArm64Main/kotlin/androidx/collection/internal/Lock.macosArm64.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    private class Resources {
+        private val arena = Arena()
+        private val attr: pthread_mutexattr_t = arena.alloc()
+        val mutex: pthread_mutex_t = arena.alloc()
+
+        init {
+            pthread_mutexattr_init(attr.ptr)
+            pthread_mutexattr_settype(attr.ptr, PTHREAD_MUTEX_RECURSIVE.convert())
```
Comment (arkivanov) [inline collection/collection/src/macosArm64Main/kotlin/androidx/collection/internal/Lock.macosArm64.kt:59]: Most likely `convert` is not need here, if duplicated per each source set.

**Pull request closed without merging**